torch/optim/sparse_adam.py

```diff
@@ -31,6 +31,18 @@ def __init__(self, params, lr=1e-3, betas=(0.9, 0.999), eps=1e-8):
             raise ValueError("Invalid beta parameter at index 0: {}".format(betas[0]))
         if not 0.0 <= betas[1] < 1.0:
             raise ValueError("Invalid beta parameter at index 1: {}".format(betas[1]))
+
+        sparse_params = []
+        for index, param in enumerate(params):
+            if isinstance(param, dict):
+                param = param.get("params")
```

Comment:
If we get a parameter dictionary, then `param` should be a List of Tensors. So instead of checking if `param` is sparse, we want to check if any of the tensors in `param` is sparse.